### PR TITLE
Automate toolchain building

### PR DESCRIPTION
**Pull request by sbogolepov**

### CODE REVIEW

File: tools/toolchain_builder/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+# Install crosstool-ng deps.
+RUN apt-get update
+RUN apt-get install -y curl gcc git g++ gperf bison flex texinfo help2man make libncurses5-dev \
+    python3-dev autoconf automake libtool libtool-bin gawk wget bzip2 xz-utils unzip \
+    patch libstdc++6 rsync
+
+# Put a fix for strip.
+COPY patches/github_pull_1244.patch .
+# Install crosstool-ng.
+RUN git clone --branch crosstool-ng-1.24.0 --depth 1 https://github.com/crosstool-ng/crosstool-ng.git && \
```
Comment (SvyatoslavScherbina) [inline tools/toolchain_builder/Dockerfile:16]: Maybe pin specific revision?

File: tools/toolchain_builder/build_zlib.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+CHOST=$TARGET \
+CC=$TOOLCHAIN_BIN_PREFIX-gcc \
+AR=$TOOLCHAIN_BIN_PREFIX-ar \
+RANLIB=$TOOLCHAIN_BIN_PREFIX-ranlib \
+./configure \
+--prefix=$INSTALL_PATH
+
+make && make install
+
+popd
```
Comment (SvyatoslavScherbina) [inline tools/toolchain_builder/build_zlib.sh:22]: Is this script used?

File: tools/toolchain_builder/run_container.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+set -eou
+
+CONTAINTER_NAME=ct-ng-kotlin
```
Comment (SvyatoslavScherbina) [inline tools/toolchain_builder/run_container.sh:4]: or maybe `toolchain_builder(_kotlin)`?

File: tools/toolchain_builder/create_image.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+docker build -t ct-ng .
```
Comment (SvyatoslavScherbina) [inline tools/toolchain_builder/create_image.sh:3]: Probably should use more unique name here, similar to `run_container.sh`.

**Review by SvyatoslavScherbina — APPROVED** [review]

**Review by PavelPunegov — APPROVED** [review]

**Pull request merged**